core/src/franchise/player/player.resolver.ts

```diff
@@ -54,4 +77,127 @@ export class PlayerResolver {
 
         return this.popService.populateOneOrFail(Player, player, "member");
     }
+
+    @Mutation(() => String)
+    async changePlayerSkillGroup(
+        @Args("playerId", {type: () => Int}) playerId: number,
+        @Args("salary", {type: () => Float}) salary: number,
+        @Args("skillGroupId", {type: () => Int}) skillGroupId: number,
+    ): Promise<string> {
+        const player = await this.playerService.getPlayer({
+            where: {id: playerId},
+            relations: {
+                member: {
+                    user: {
+                        authenticationAccounts: true,
+                    },
+                    organization: true,
+                    profile: true,
+                },
+                skillGroup: {
+                    organization: true,
+                    game: true,
+                    profile: true,
+                },
+            },
+        });
+
+        const skillGroup = await this.skillGroupService.getGameSkillGroup({
+            where: {
+                id: skillGroupId,
+            },
+            relations: {
+                profile: true,
+            },
+        });
+
+        const discordAccount = await this.userAuthRepository.findOneOrFail({
+            where: {
+                user: {
+                    id: player.member.user.id,
+                },
+                accountType: UserAuthenticationAccountType.DISCORD,
+            },
+        });
+        const orgProfile = await this.organizationService.getOrganizationProfileForOrganization(player.member.organization.id);
+
+        if (player.skillGroup.id === skillGroupId) return "ERROR: You are already in this skill group";
+
+        const inputData: ManualSkillGroupChange = {
+            id: playerId,
+            salary: salary,
+            skillGroup: skillGroup.ordinal,
+        };
+        
+        await this.playerService.updatePlayerStanding(playerId, salary, skillGroupId);
+        await this.playerService.mle_rankDownPlayer(playerId, salary);
+        await this.eloConnectorService.createJob(EloEndpoint.SGChange, inputData);
+
+        await this.eventsService.publish(EventTopic.PlayerSkillGroupChanged, {
+            playerId: player.id,
+            name: player.member.profile.name,
+            organizationId: player.skillGroup.organizationId,
+            discordId: discordAccount.accountId,
+            old: {
+                id: player.skillGroup.id,
+                name: player.skillGroup.profile.description,
+                salary: Number(player.salary),
+                discordEmojiId: player.skillGroup.profile.discordEmojiId,
+            },
+            new: {
+                id: skillGroup.id,
+                name: skillGroup.profile.description,
+                salary: Number(salary),
+                discordEmojiId: skillGroup.profile.discordEmojiId,
+            },
+        });
+
+        await this.notificationService.send(NotificationEndpoint.SendNotification, {
```

Comment:
Should this be handled whenever a rank out occurs?

Reply:
We currently get DM notifications from emilio whenever LO takes direct action, like for a manual rankout. I don't see why we wouldn't continue with that behavior.